### ISSUE TITLE
`Experiment.fetch_data` raises error if using `CustomData`

### ISSUE THREAD

**Issue by FriendLey** [facebook/Ax#317]
The Type of `Data` columns only support: 
```
COLUMN_DATA_TYPES = {
    "arm_name": str,
    "metric_name": str,
    "mean": np.float64,
    "sem": np.float64,
    "trial_index": np.int64,
    "start_time": TPdTimestamp,
    "end_time": TPdTimestamp,
    "n": np.int64,
    "frac_nonnull": np.float64,
    "random_split": np.int64,
    "fidelities": str,  # Dictionary stored as json
}
``` 
What should I do if I want to save other column type in Data for subsequent use?

**Comment by Jakepodell** [comment]
Hi @FriendLey!

We have the following function to create a custom data class: https://github.com/facebook/Ax/blob/master/ax/core/data.py#L263

Please let us know if this helps!

**Comment by FriendLey** [comment]
@Jakepodell Thanks for your response. 

If Using `CustomData`, The `fetch_data()` function of `Experiment` will error because it uses `Data.from_multiple_data()` function. Does it mean that we must override `fetch_data` in a subclass of `Experiment` when using `CustomData` class? This is not elegant if so.

**Comment by Jakepodell** [comment]
Hi!

Thanks for pointing this out- this is something that is on our queue of bugs to fix. In the meantime, you can subclass experiment as you mentioned, or write your own data fetching function and use [Experiment.attach_data] (https://github.com/facebook/Ax/blob/master/ax/core/experiment.py#L483).

Here is a working example of `CustomData.from_multiple_data` from one of our other projects: 
```
SchedulerData: Type[Data] = custom_data_class(required_columns={"trial_index"})
...
...
...
return SchedulerData.from_multiple_data(
            [
                self.fetch_trial_data(trial=trial)
                for trial in self.experiment.get_trials_by_indices(
                    trial_indices=trial_indices
                )
            ]
        )
```

Please let us know if you run into any other issues related to the custom_data_class, which we are working hard to better support.

**Comment by lena-kashtelyan** [comment]
@FriendLey, just a quick follow-up to your question here. How were you looking to use a custom `Data` class? Were you going to return an instance of one from your metric's `fetch_data` method? It would help us move forward with the fix to know approximately what your use case was.

**Comment by FriendLey** [comment]
> @FriendLey, just a quick follow-up to your question here. How were you looking to use a custom `Data` class? Were you going to return an instance of one from your metric's `fetch_data` method? It would help us move forward with the fix to know approximately what your use case was.

Yes, I returned an instance of the custom `Data` class from metric's `fetch_data` method which contains columns not in COLUMN_DATA_TYPES. But the class Experiment doesn't support it because it's fetch_data use Data.from_multiple_data().

**Comment by lena-kashtelyan** [comment]
@FriendLey, the fix for this is now on `master` and will be included in the next stable version we release!

**Comment by lena-kashtelyan** [comment]
The fix is now in the latest stable, 0.1.18.
